Yoast/wordpress-seo
# Escaped the error message value to prevent xss scripting

## Context


* We want to escape the error message value of the social tab fields to prevent XSS scripting

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where XSS scripting could be done in the social fields when entering something like `x3<img src=a onerror=alert(/Facebook-App-ID/)>%*|{}!#$^*&e3\`.

## Relevant technical choices:

* Spoke with @moorscode and validated with `esc_html`

## Test instructions

This PR can be tested by following these steps:

For Twitter Username:
- Checkout this branch (xss-social-scripting-escaping), build and activate the plugin

- Navigate to "SEO"

- Navigate to "Social" sub-menu

- Set "Twitter Username" to:
`x3<img src=a onerror=alert(/Facebook-App-ID/)>%*|{}!#$^*&e3\`

- Click "Save Changes"

- Note that no popup appears

- Repeat these steps on trunk and make sure the popup **does appear**

For Facebook App ID:
- Checkout this branch (xss-social-scripting-escaping), build and activate the plugin

- Navigate to "SEO"

- Navigate to "Social" sub-menu

- Navigate to "Facebook" tab

- Set "Facebook App ID" to:
`x3<img src=a onerror=alert(/Facebook-App-ID/)>%*|{}!#$^*&e3\`

- Click "Save Changes"

- Note that no popup appears

- Repeat these steps on trunk and make sure the popup **does appear**



## UI changes

## Documentation

## Quality assurance


Fixes #1033
